WebXR/systems/sys_transform.ts

```diff
@@ -8,42 +8,38 @@ const QUERY = Has.Transform;
 export function sys_transform(game: Game, delta: number) {
     for (let i = 0; i < game.World.Signature.length; i++) {
         if ((game.World.Signature[i] & QUERY) === QUERY) {
-            update(game, i);
+            let transform = game.World.Transform[i];
```

Comment:
You need to fetch every transform component to check the dirty flag. Whereas adding a dirty signature would avoid this.

Reply:
Thanks for bringing this up; I should have written this down somewhere and you question gives me an opportunity to do it.

Like I mentioned in #34, I do think a separate `Has.Dirty` signature is a more pure approach and in fact I thought back to your implementation when I was writing this code. I considered it here but ultimately decided against it for two reasons.

1. I didn't want to inflate the number of components.
2. I slightly prefer the developer experience of `transform.Dirty`, mostly for its convenience.

For (2), compare the following snippets which move the entity 1 unit to the left:

```ts
transform.Position[0] += 1;
transform.Dirty = true
```

```ts
transform.Position[0] += 1;
world.Signature[entity] |= Has.Dirty;
```

Like I mentioned in #34, to me `transform.Dirty` is a more pragmatic choice, even if it might be slightly less performant.

Reply:
You could implement a [bitset](https://github.com/nim-lang/Nim/blob/devel/compiler/bitsets.nim) using an array that way you wouldn't have any restrictions.